org.eclipse.scout.rt.rest.test/src/test/java/org/eclipse/scout/rt/rest/client/proxy/ErrorDoRestClientExceptionTransformerTest.java

```diff
@@ -0,0 +1,128 @@
+    this.assertVetoException(ResourceNotFoundException.class, mockCause, mockTitle, mockCode, m_exceptionTransformer.transform(mockCause, mockResponse(Status.NOT_FOUND, errorResponse)));
+
+    // Client errors -  commonly handled (4xx)
+    assertVetoException(VetoException.class, mockCause, mockTitle, mockCode, m_exceptionTransformer.transform(mockCause, mockResponse(Status.BAD_REQUEST, errorResponse)));
+    assertVetoException(VetoException.class, mockCause, mockTitle, mockCode, m_exceptionTransformer.transform(mockCause, mockResponse(Status.UNAUTHORIZED, errorResponse)));
+    assertVetoException(VetoException.class, mockCause, mockTitle, mockCode, m_exceptionTransformer.transform(mockCause, mockResponse(Status.CONFLICT, errorResponse)));
+    assertVetoException(VetoException.class, mockCause, mockTitle, mockCode, m_exceptionTransformer.transform(mockCause, mockResponse(Status.REQUEST_HEADER_FIELDS_TOO_LARGE, errorResponse)));
+
+    // Client errors - commonly handled without error response (4xx)
+    assertVetoException(VetoException.class, mockCause, null, null, m_exceptionTransformer.transform(mockCause, mockResponse(Status.BAD_REQUEST)));
+
+    // Server errors - specially handled (5xx)
+    assertException(RemoteSystemUnavailableException.class, mockCause, m_exceptionTransformer.transform(mockCause, mockResponse(Status.BAD_GATEWAY)));
+    assertException(RemoteSystemUnavailableException.class, mockCause, m_exceptionTransformer.transform(mockCause, mockResponse(Status.SERVICE_UNAVAILABLE)));
+    assertException(RemoteSystemUnavailableException.class, mockCause, m_exceptionTransformer.transform(mockCause, mockResponse(Status.GATEWAY_TIMEOUT)));
+
+    // Server errors -  commonly handled (5xx)
+    assertException(ProcessingException.class, mockCause, m_exceptionTransformer.transform(mockCause, mockResponse(Status.INTERNAL_SERVER_ERROR)));
+    assertProcessingException(ProcessingException.class, mockCause, mockTitle, mockCode, m_exceptionTransformer.transform(mockCause, mockResponse(Status.INTERNAL_SERVER_ERROR, errorResponse)));
+  }
+
+  protected Response mockResponse(Response.Status status) {
+    return mockResponse(status, null);
+  }
+
+  protected Response mockResponse(Response.Status status, ErrorResponse errorResponse) {
+    Response res = Mockito.mock(Response.class);
+    when(res.getStatus()).thenReturn(status.getStatusCode());
+    when(res.getStatusInfo()).thenReturn(status);
+    when(res.readEntity(ErrorResponse.class)).thenReturn(errorResponse);
+    when(res.hasEntity()).thenReturn(errorResponse != null);
+    return res;
+  }
+
+  protected void assertException(Class<?> expected, Throwable cause, RuntimeException actual) {
+    assertInstance(actual, expected);
+    assertEquals(cause, actual.getCause());
+  }
+
+  protected void assertVetoException(Class<? extends VetoException> expected, Throwable expectedCause, String expectedTitle, Integer expectedCode, RuntimeException actual) {
```

Comment:
Could be merged with `assertProcessingException` because VetoException extends ProcessingException.